src/resource/package_builder.rs

```diff
@@ -0,0 +1,1007 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io;
+use std::io::{Read, Seek, Write};
+use std::path::{Path, PathBuf};
+
+use crate::resource::pdefs::{PartitionId, PartitionType};
+use crate::resource::resource_package::{
+    ChunkType, PackageHeader, PackageMetadata, PackageOffsetFlags, PackageOffsetInfo,
+    PackageVersion, ResourceHeader, ResourcePackage, ResourcePackageSource,
+    ResourceReferenceCountAndFlags, ResourceReferenceFlags,
+};
+use crate::resource::resource_partition::PatchId;
+use crate::resource::runtime_resource_id::RuntimeResourceID;
+use crate::{GlacierResource, GlacierResourceError, WoaVersion};
+use binrw::BinWrite;
+use binrw::__private::Required;
+use binrw::io::Cursor;
+use binrw::meta::WriteEndian;
+use indexmap::{IndexMap, IndexSet};
+use lzzzz::{lz4, lz4_hc};
+use thiserror::Error;
+
+/// `PackageResourceBlob` is an enum representing various types of package resource stores, which can 
+/// include files, file sections, and memory buffers, optionally compressed or scrambled.
+enum PackageResourceBlob {
```

Comment:
`should_scramble` and `is_scrambled` are probably relevant on all blob variants, if `is_scrambled` is false, we might still want to scramble it anyway.